extension.js

```diff
@@ -36,34 +36,40 @@ exports.activate = function activate(context) {
 function createSnippet(snippetFilename, editor) {
     fs.readFile(snippetFilename, 'utf8', function(e, data) {
         console.log(data);
-        var jsonData = json.parse(data, null, false);
-        jsonData = addSelectedSnippet(jsonData, editor);
-        updateSnippetFile(snippetFilename, jsonData);
+        var jsonData = (data ? json.parse(data, null, false) : {});
+        addSelectedSnippetAsync(jsonData, editor).then(function(result) {
+            jsonData = result;
+            updateSnippetFile(snippetFilename, jsonData);
+        });
     });
 }
 
-function addSelectedSnippet(jsonData, editor) {
+function addSelectedSnippetAsync(jsonData, editor) {
     var newJsonData = {};
 
     var snippetBody = getSnippetBody(editor);
 
-    if (typeof (jsonData['//$']) !== 'undefined') {
-        // newJsonData['//$'] = jsonData['//$'];
-    }
-
-    newJsonData[DEFAULT_SNIPPET_NAME] = {
-        prefix: 'yourPrefixHere',
-        body: snippetBody,
-        description: 'Your snippet description here.'
-    };
+    return vscode.window.showInputBox({prompt:'Enter a name for your snippet'}).then(function(value) {
```

Comment:
Anonymous functions are annoying to debug & navigate, worth pulling it out into a separate method.